### ISSUE TITLE
How to use the pretrained weight of CoOp for Multi-label Top-K Prediction?

### ISSUE THREAD

**Issue by QingchenWait** [KaiyangZhou/CoOp#56]
Thanks for sharing the code.
I wish to apply the pretrained weights to the Multi-label prediction task. **I would like to input the optimized prompt and 1 picture to the frozen CLIP encoders to get the TOP-K prediction results**.
I followed the overview structure picture to code for the task. However, i met some problems in it.

In the pretrained weight, the prompt token [V1][V2]... has a shape of [n_ctx,512]. However, it does not match the normal prompt token with [CLS] (shape of [n_cls,77]), which means it cannot concat directly.
I saw the token_prefix and token_suffix embeddings in the weight file. After concat them with prompt_token, it gives a 3d tensor with the shape [n_cls,n_ctx,512], which does not match the image feature.

So, I would like to know that **how to adjust the size of the optimized prompt and [CLASS] token so that it can be used as the input of CLIP's text encoder?**

I am looking forward to the reply. Thanks a lot if you could give the help!

**Comment by QingchenWait** [comment]
Hi, i am happy to tell you that I have solved my problem. The classifier performs well in my task. 
Thanks for the job!

**Comment by QingchenWait** [comment]
Hi, i am happy to tell you that I have solved my problem. The classifier performs well in my task.
Thanks for the job!

**Comment by Xucy233** [comment]
Hello, have you apply on NUSWIDE?I want to know how to apply it to NUS-WIDE,thanks!

**Comment by QingchenWait** [comment]
> Hello, have you apply on NUSWIDE?I want to know how to apply it to NUS-WIDE,thanks!

Hi, although i did not apply CoOp on NUSWIDE, but i did try on my own classification dataset. 
The overall process is a little complicated. To sum up, Firstly, you need to preprocess the samples in the dataset to combine the image path, the corresponding category name and ID into a file with a format similar to split_zhou_xxxx.json (It can be seen in [DATASETS.md](https://github.com/KaiyangZhou/CoOp/blob/main/DATASETS.md) ). Secondly, you need to code the DataLoader following the requirement in DASSL. (https://github.com/KaiyangZhou/Dassl.pytorch#add-a-dataset)

**Comment by zhangyupeng123** [comment]
Hi~Do you have the original COOP model weights, not the ones for 4/16 shots?
